### PR TITLE
[ALLI-6789] LIDO: index preferred title

### PR DESCRIPTION
**Pull request by LuomaJuha**

### CODE REVIEW

File: src/RecordManager/Base/Record/Lido.php [hunk 1]
```diff
@@ -237,10 +237,22 @@ public function getTitle(
...
+                if (!($title = trim((string)$appellationValue))) {
+                    continue;
+                }
+                $allTitles[] = $title;
+                if (!$lang) {
+                    continue;
+                }
+                $priority = (string)$appellationValue['pref'];
+                $titleLang = (string)$appellationValue['lang'];
+                if (('preferred' === $priority && $titleLang === $lang)
```
Comment (EreMaijala) [inline src/RecordManager/Base/Record/Lido.php]: I think you could simplify the conditions quite a bit by giving these defaults:
```
priority = (string)($appellationValue['pref'] ?? 'preferred');
$titleLang = (string)($appellationValue['lang'] ?? $lang);
```

File: src/RecordManager/Base/Record/Lido.php [hunk 2]
```diff
@@ -237,10 +237,22 @@ public function getTitle(
         $allTitles = [];
         foreach ($this->getTitleSetNodes() as $set) {
             foreach ($set->appellationValue as $appellationValue) {
-                if ($lang == null || $appellationValue['lang'] == $lang) {
-                    $titles[] = (string)$appellationValue;
+                if (!($title = trim((string)$appellationValue))) {
+                    continue;
+                }
+                $allTitles[] = $title;
+                if (!$lang) {
```
Comment (EreMaijala) [inline src/RecordManager/Base/Record/Lido.php]: This makes the loop work differently than before where we took the value even when $lang was null. Sure this works right?

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: src/RecordManager/Base/Record/Lido.php (hunk 1 above)
Comment (LuomaJuha) [reply src/RecordManager/Base/Record/Lido.php]: Casting is a habit when working with SimpleXmlElements... Will adjust.

File: src/RecordManager/Base/Record/Lido.php (hunk 2 above)
Comment (LuomaJuha) [reply src/RecordManager/Base/Record/Lido.php]: Oh yeah, if there is a lang and a no lang in the same set then its not going to be the same. I think I could actually add the no lang version also to the list.

File: src/RecordManager/Base/Record/Lido.php (hunk 1 above)
Comment (EreMaijala) [reply src/RecordManager/Base/Record/Lido.php]: ...And I was wrong, the cast is needed.

Comment (EreMaijala) [reply src/RecordManager/Base/Record/Lido.php]: Updated the example to reflect that.

File: src/RecordManager/Base/Record/Lido.php [hunk 3]
```diff
@@ -220,27 +220,32 @@ public function toSolrArray(Database $db = null)
...
+     * @param array  $excludedDescriptions Description types to exclude
      *
      * @return string
      */
     public function getTitle(
-        $forFiling = false,
-        $lang = null,
-        $excludedDescriptions = ['provenance']
-    ) {
+        bool $forFiling = false,
```
Comment (EreMaijala) [inline src/RecordManager/Base/Record/Lido.php]: If you add the types here, you'll need to do it for the base class and all other derivatives as well (see the failed tests). Better to do that in a separate refactoring task that does not change functionality.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**